internal/handlers/common/aggregations/operators/accumulators/accumulators.go

```diff
@@ -0,0 +1,81 @@
+// Copyright 2021 FerretDB Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package accumulators provides aggregation accumulator operators.
+// Accumulators are different from other operators as they perform operations
+// on a group of documents rather than a single document.
+// They are used only in small subset of available stages (like `$group`).
+//
+// Accumulators that can be used outside of accumulation with different behaviour (like `$sum`),
+// should be stored in both operators and accumulators packages.
+package accumulators
+
+import (
+	"fmt"
+
+	"github.com/FerretDB/FerretDB/internal/handlers/commonerrors"
+	"github.com/FerretDB/FerretDB/internal/types"
+)
+
+// newAccumulatorFunc is a type for a function that creates an accumulation operator.
+type newAccumulatorFunc func(expression *types.Document) (Accumulator, error)
+
+// Accumulator is a common interface for aggregation accumulation operators.
+type Accumulator interface {
+	// Accumulate documents and returns the result of applying operator.
+	Accumulate(in []*types.Document) (any, error)
```

Comment:
Do we work with slices (not iterators) because for accumulators we need all the documents anyway?

Reply:
Yes exactly! Providing iterator here doesn't make a lot of sense as we already store them in memory. We might move to iterators in the future if we change how stages organize the data though.

Reply:
Nope